xbmc/addons/AddonInstaller.cpp

```diff
@@ -734,7 +751,7 @@ bool CAddonInstallJob::Install(const std::string &installFrom, const RepositoryP
       const AddonVersion& version = it->version;
       bool optional = it->optional;
       AddonPtr dependency;
-      bool haveAddon = CServiceBroker::GetAddonMgr().GetAddon(addonID, dependency, ADDON_UNKNOWN, false);
+      bool haveAddon = addonMgr.GetAddon(addonID, dependency, ADDON_UNKNOWN, false);
```

Comment:
`bool haveInstalledAddon`. I think `GetAddon()` will only find installed addons correct? Maybe that function should be renamed as a separate commit.

Reply:
depends on the parameters for GetAddon i think, but will check back

Reply:
we can decide if we keep [cleanup] as `haveAddon` is used here too (`CheckDependencies`)
https://github.com/xbmc/xbmc/blob/641f0e3a73c96565fe37a90066e7c30ddfad6a91/xbmc/addons/AddonInstaller.cpp#L316-L317

Reply:
I think you can change it there too as it's the same function call.

What about renaming `GetAddon` to `GetInstalledAddon`? You could do that in the 4th commit also and call it: `[addons] refactor GetAddon function to GetInstalledAddon to aid readability`

Reply:
this would require a wrapper for `GetAddon` we cannot rename it as it's used widespread across the codebase if i see it correct

Reply:
Why not rename it everywhere?

Reply:
makes sense to rename and is scheduled

Reply:
we better leave it as is... it would clutter the pr